From 8c770088b3c7f1f1d8c5cfcb3dbb2a1e157c911f Mon Sep 17 00:00:00 2001
From: Pulkit077 <pulkitagarwal2k1@gmail.com>
Date: Sat, 22 May 2021 19:02:41 +0530
Subject: [PATCH 27/27] SystemUI: Blurs: allow blur even when GFX acceleration
 is disabled

 isHighEndGfx returns false when ro.config.avoid_gfx_accel or config_avoidGfxAccel is set causing to disable blur.
 Blur is still heavy but with new renderengine patches by Kdrag0n and reducing amount of blur, it seems to be less aggressive on CPU.
 With this we can still reduce memory usage and keep using blur giving users option to enable it.

 New behaviour: blurs are only disabled if device has low ram prop set.

Test: boot, enable blur toggle under developer option, reboot and blurs are visible.
Change-Id: Id1729aca89b6dff22c865ae913da044c0e7c7f31
Signed-off-by: Pulkit077 <pulkitagarwal2k1@gmail.com>
---
 .../src/com/android/systemui/shared/system/BlurUtils.java     | 2 +-
 .../SystemUI/src/com/android/systemui/statusbar/BlurUtils.kt  | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/packages/SystemUI/shared/src/com/android/systemui/shared/system/BlurUtils.java b/packages/SystemUI/shared/src/com/android/systemui/shared/system/BlurUtils.java
index 9f26d851f77..87a6c9d8780 100644
--- a/packages/SystemUI/shared/src/com/android/systemui/shared/system/BlurUtils.java
+++ b/packages/SystemUI/shared/src/com/android/systemui/shared/system/BlurUtils.java
@@ -32,6 +32,6 @@ public abstract class BlurUtils {
      * @return {@code true} when supported.
      */
     public static boolean supportsBlursOnWindows() {
-        return mBlurSupportedSysProp && !mBlurDisabledSysProp && ActivityManager.isHighEndGfx();
+        return mBlurSupportedSysProp && !mBlurDisabledSysProp && !ActivityManager.isLowRamDeviceStatic();
     }
 }
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/BlurUtils.kt b/packages/SystemUI/src/com/android/systemui/statusbar/BlurUtils.kt
index c70d3847bd6..e6cae5b5417 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/BlurUtils.kt
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/BlurUtils.kt
@@ -99,7 +99,7 @@ open class BlurUtils @Inject constructor(
      * @return {@code true} when supported.
      */
     open fun supportsBlursOnWindows(): Boolean {
-        return blurSupportedSysProp && !blurDisabledSysProp && ActivityManager.isHighEndGfx()
+        return blurSupportedSysProp && !blurDisabledSysProp && !ActivityManager.isLowRamDeviceStatic()
     }
 
     override fun dump(fd: FileDescriptor, pw: PrintWriter, args: Array<out String>) {
@@ -113,4 +113,4 @@ open class BlurUtils @Inject constructor(
             it.println("supportsBlursOnWindows: ${supportsBlursOnWindows()}")
         }
     }
-}
\ No newline at end of file
+}
-- 
2.31.1.606.gdf6c4f722c

